BungieFrameWork/BFW_Source/BFW_LocalInput/Platform_SDL/BFW_LI_Platform_SDL.c

```diff
@@ -0,0 +1,630 @@
+	
+	{LIcKeyCode_NumLock, SDLK_NUMLOCKCLEAR},
+	{LIcKeyCode_Divide, SDLK_KP_DIVIDE},
+	{LIcKeyCode_Multiply, SDLK_KP_MULTIPLY},
+	{LIcKeyCode_Subtract, SDLK_KP_MINUS},
+	{LIcKeyCode_Add, SDLK_KP_PLUS},
+	{LIcKeyCode_Decimal, SDLK_KP_DECIMAL},
+	{LIcKeyCode_NumPadEnter, SDLK_KP_ENTER},
+	{LIcKeyCode_NumPadComma, SDLK_KP_COMMA},
+	{LIcKeyCode_NumPadEquals, SDLK_KP_EQUALS},
+	
+	{LIcKeyCode_NumPad0, SDLK_KP_0},
+	{LIcKeyCode_NumPad1, SDLK_KP_1},
+	{LIcKeyCode_NumPad2, SDLK_KP_2},
+	{LIcKeyCode_NumPad3, SDLK_KP_3},
+	{LIcKeyCode_NumPad4, SDLK_KP_4},
+	{LIcKeyCode_NumPad5, SDLK_KP_5},
+	{LIcKeyCode_NumPad6, SDLK_KP_6},
+	{LIcKeyCode_NumPad7, SDLK_KP_7},
+	{LIcKeyCode_NumPad8, SDLK_KP_8},
+	{LIcKeyCode_NumPad9, SDLK_KP_9},
+
+	{LIcKeyCode_LeftWindowsKey, SDLK_LGUI},
+	{LIcKeyCode_RightWindowsKey, SDLK_RGUI},
+};
+
+// assert that ASCII special characters match
+#if __STDC_VERSION__ >= 201112L
+_Static_assert(
+	SDLK_AMPERSAND    == LIcKeyCode_Ampersand    &&
+	SDLK_ASTERISK     == LIcKeyCode_Star         &&
+	SDLK_AT           == LIcKeyCode_At           &&
+	SDLK_BACKQUOTE    == LIcKeyCode_Grave        &&
+	SDLK_BACKSLASH    == LIcKeyCode_BackSlash    &&
+	SDLK_BACKSPACE    == LIcKeyCode_BackSpace    &&
+	SDLK_CARET        == LIcKeyCode_Hat          &&
+	SDLK_COLON        == LIcKeyCode_Colon        &&
+	SDLK_COMMA        == LIcKeyCode_Comma        &&
+	SDLK_DOLLAR       == LIcKeyCode_Dollar       &&
+	SDLK_EQUALS       == LIcKeyCode_Equals       &&
+	SDLK_ESCAPE       == LIcKeyCode_Escape       &&
+	SDLK_EXCLAIM      == LIcKeyCode_Exclamation  &&
+	SDLK_GREATER      == LIcKeyCode_GreaterThan  &&
+	SDLK_HASH         == LIcKeyCode_Pound        &&
+	SDLK_LEFTBRACKET  == LIcKeyCode_LeftBracket  &&
+	SDLK_LEFTPAREN    == LIcKeyCode_LeftParen    &&
+	SDLK_LESS         == LIcKeyCode_LessThan     &&
+	SDLK_MINUS        == LIcKeyCode_Minus        &&
+	SDLK_PERCENT      == LIcKeyCode_Percent      &&
+	SDLK_PERIOD       == LIcKeyCode_Period       &&
+	SDLK_PLUS         == LIcKeyCode_Plus         &&
+	SDLK_QUESTION     == LIcKeyCode_Question     &&
+	SDLK_QUOTE        == LIcKeyCode_Apostrophe   &&
+	SDLK_QUOTEDBL     == LIcKeyCode_Quote        &&
+	SDLK_RETURN       == LIcKeyCode_Return       &&
+	SDLK_RIGHTBRACKET == LIcKeyCode_RightBracket &&
+	SDLK_RIGHTPAREN   == LIcKeyCode_RightParen   &&
+	SDLK_SEMICOLON    == LIcKeyCode_Semicolon    &&
+	SDLK_SLASH        == LIcKeyCode_Slash        &&
+	SDLK_SPACE        == LIcKeyCode_Space        &&
+	SDLK_TAB          == LIcKeyCode_Tab          &&
+	SDLK_UNDERSCORE   == LIcKeyCode_Underscore   ,
+	"Keys do not match"
+);
+#endif
+
+static const unsigned num_sdl_keys = sizeof(sdl_keyboard_translation_table)/sizeof(struct sdl_oni_key);
+
+static SDL_Keycode oni_to_sdl_keycode(LItKeyCode key)
+{
+	int i;
+	
+	if (key < 0x80)
+	{
+		// ascii range maps directly
+		// except...
+		switch (key)
+		{
+			case LIcKeyCode_Tilde:
+			case LIcKeyCode_LeftCurlyBracket:
+			case LIcKeyCode_RightCurlyBracket:
```

Comment:
On a US keyboard `[` and `{` are on the same key -> `SDL_SCANCODE_LEFTBRACKET` and `SDL_SCANCODE_RIGHTBRACKET`